src/riscv-legacy-integration.adoc

```diff
@@ -55,26 +56,27 @@ top is 2^MXLEN^
 [#section-cheri-execution-mode]
 === CHERI Execution Mode Encoding
 
-{cheri_legacy_ext_name} adds a new CHERI execution mode bit (M) to
-capabilities. In RV32, the mode is encoded in the permissions field as
-described in xref:cap_permissions_encoding[xrefstyle=short]. In RV64, a new
-mode bit field is added to the capability format as shown in
-xref:cap_encoding_xlen64_mode[xrefstyle=short]. The current CHERI execution
-mode is given by the M bit of the <<pcc>> and the CRE bits in <<mseccfg>>,
-<<menvcfg>>, and <<senvcfg>> as follows:
+{cheri_legacy_ext_name} adds a new CHERI execution mode field (M) to
+capabilities within the AP field (see xref:section_cap_perms[xrefstyle=short]).
+When MXLEN=32, the mode is encoded in bit 0 of quadrant 1 from the AP field.
+When MXLEN=64, the mode is encoded in bit 0 of the AP fielda new M bit field is
+added to the capability format as shown in
+xref:cap_perms_encoding64[xrefstyle=short]. In both encodings, M=0
+indicates Legacy Mode and M=1 indicates Capability Mode.
 
-* The mode is Capability when the M bit of the <<pcc>> is 1 and the effective
-CRE=1 for the current privilege level
-* The mode is Legacy when the effective CRE=0 for the current privilege level or the M bit of the <<pcc>> is 0
+The current CHERI execution mode is given by the M field of the <<pcc>> and the
+CRE bits in <<mseccfg>>, <<menvcfg>>, and <<senvcfg>> as follows:
 
-.Capability encoding when MXLEN=64 and {cheri_legacy_ext_name} is supported
-[#cap_encoding_xlen64_mode]
-include::img/cap-encoding-xlen64-mode.edn[]
+* The mode is Capability when the M field of the <<pcc>> is Capability and the effective
+CRE=1 for the current privilege level
+* The mode is Legacy when the effective CRE=0 for the current privilege level or the M field of the <<pcc>> is Legacy
 
-For RV64, the M bit can be set to 1 when the capability does not grant <<x_perm>>.
-In this case, the M bit is superfluous, so the encoding may be used to support additional features in future extensions.
+When MXLEN=32, the M field is Legacy Mode in the <<infinite-cap>> capability,
+so its AP field is 8. The <<null-cap>> capability does not grant <<x_perm>>, so
+the M field is superfluous in this case.
 
-The M bit is 0 in both the <<null-cap>> and <<infinite-cap>> capabilities.
+When MXLEN=64, the M bit can be set to 1 when the capability does not grant <<x_perm>>.
```

Comment:
I'd suggest putting the MXLEN=64 paragraph first since I was confused why we are talking about the null cap when reading it in order, but it made sense after reading the 64-bit one.

Reply:
The NULL part only applies to MXLEN=32, so its really a continuation of the same idea. I will change the punctuation to make it clearer.